### PR TITLE
chore(chat-sdk): adapt fetcher options for gql query

### PR DESCRIPTION
**Pull request by liangfung**

### CODE REVIEW

File: ee/tabby-ui/lib/tabby/token-management.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import { jwtDecode } from 'jwt-decode'
 import { isNil } from 'lodash-es'
+import { FetcherOptions } from 'tabby-chat-panel/index'
 
 import { isClientSide } from '../utils'
 import { AuthData } from './auth'
 
 export const AUTH_TOKEN_KEY = '_tabby_auth'
 export const AUTH_LOCK_KEY = '_tabby_auth_lock'
+export const FETCHER_OPTIONS_KEY = '_tabby_fetcher_options'
```
Comment (wsxiaoys) [inline ee/tabby-ui/lib/tabby/token-management.ts:10]:
```suggestion
export const FETCHER_OPTIONS_KEY = '_tabby_chat_sdk_fetcher_options'
```

File: ee/tabby-ui/lib/tabby/token-management.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { jwtDecode } from 'jwt-decode'
 import { isNil } from 'lodash-es'
+import { FetcherOptions } from 'tabby-chat-panel/index'
 
 import { isClientSide } from '../utils'
 import { AuthData } from './auth'
 
 export const AUTH_TOKEN_KEY = '_tabby_auth'
 export const AUTH_LOCK_KEY = '_tabby_auth_lock'
+export const FETCHER_OPTIONS_KEY = '_tabby_chat_sdk_fetcher_options'
```
Comment (wsxiaoys) [inline ee/tabby-ui/lib/tabby/token-management.ts:10]:
```suggestion
const FETCHER_OPTIONS_KEY = '_tabby_chat_sdk_fetcher_options'
```

File: ee/tabby-ui/app/chat/page.tsx [hunk 3]
```diff
@@ -170,6 +172,10 @@ export default function ChatPage() {
       if (chatRef.current) return
       setActiveChatId(nanoid())
       setIsInit(true)
+      // save fetcherOptions to session storage
+      if (client) {
```
Comment (liangfung) [inline ee/tabby-ui/app/chat/page.tsx:176]: cc @icycodes

File: ee/tabby-ui/app/chat/page.tsx
Comment (wsxiaoys) [inline ee/tabby-ui/app/chat/page.tsx:1]: https://github.com/TabbyML/tabby/blob/a1da9012be176c50bf71934eddd8c302eb329e71/ee/tabby-ui/app/chat/page.tsx#L336

this headers can be removed now?

Comment (liangfung) [reply ee/tabby-ui/app/chat/page.tsx:1]: done

**Pull request merged**